src/Big5Utils/Big5Utils.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2023 and onwards The McBopomofo Authors.
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+#include "Big5Utils.h"
+
+#include <unicode/ucnv.h>
+
+#include <cstdint>
+
+namespace Big5Utils {
+
+bool isValidSingleUtf8Character(const char* str, int32_t length) {
+  if (str == nullptr || length <= 0) {
+    return false;
+  }
+  const uint8_t* s = reinterpret_cast<const uint8_t*>(str);
+  int32_t i = 0;
+  UChar32 c;
+  U8_NEXT(s, i, length, c);
+  if (c < 0) {  // Invalid sequence
+    return false;
+  }
+  return i == length;  // Check if we consumed the whole string
+}
+
+std::string convertBig5fromUint16(uint16_t codePoint) {
+  UErrorCode status = U_ZERO_ERROR;
+  UConverter* conv = ucnv_open("Big5", &status);
```

Comment:
I wonder if you would want to use `windows-950-2000`, which is the canonical name of the converter (see https://icu4c-demos.unicode.org/icu-bin/convexp?conv=big5). This also makes it clear that this does not intend to support Big5- HKSCS.